scripts/generate-workloaddependencies-json/generate-workloaddependencies-json.cs

```diff
@@ -0,0 +1,39 @@
+// arguments are: <platform> <version> <xcodeVersion> <sdkVersion> <outputPath>
+
+using System.IO;
+using System.Xml;
+
+var expectedArgumentCount = 5;
+if (args.Length != expectedArgumentCount) {
+	Console.WriteLine ($"Need {expectedArgumentCount} arguments, got {args.Length}");
+	return 1;
+}
+
+var argumentIndex = 0;
+var platform = args [argumentIndex++];
+var version = args [argumentIndex++];
+var xcodeVersion = args [argumentIndex++];
+var sdkVersion = args [argumentIndex++];
+var outputPath = args [argumentIndex++];
+
+var platformLowerCase = platform.ToLowerInvariant ();
+
+using (var writer = new StreamWriter (outputPath)) {
+	writer.WriteLine ($"{{");
+	writer.WriteLine ($"  \"microsoft.net.sdk.{platformLowerCase}\": {{");
```

Comment:
I think we do want the actual workload identifier here as is in the PR.  The aliases were meant to capture the name we might use in the CLI, but for looking up this type of info, for example, `C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.ios\[version]\WorkloadDependencies.json` is where i'd expect to find the info, so I think it makes sense for the key in this file to match the id here.

Reply:
How would we define one workload as having different requirements as the other?

For example, here in .NET 9 Preview 6:

* https://github.com/dotnet/android/blob/release/9.0.1xx-preview6/src/Xamarin.Android.Build.Tasks/Microsoft.NET.Sdk.Android/WorkloadManifest.in.json

We would require API-34 for the `android` workload and API-35 for the preview `android-35` workload.

Reply:
I think we've mostly sorted this out for now.  Android is listing these as optional, and we'll eventually add a way to query the context for a given project state (probably invoking a specific msbuild target on a project) at which point having a `net9.0-android35.0` TFM for example could cause the returned info to require the sdk package for android 35 instead of the default 34 for `net9.0` as per the file in the workload.

In the case of macios, I don't think this will be a problem as the workload doesn't differentiate anywhere at this point that different xcode versions would be needed for different target platform versions for example.